### PR TITLE
Fixed currently playing sounds volume change, disable and background

### PR DESCRIPTION
**Pull request by paezao**
There's something annoying about changing Sound options in the current ClassicUO release.

When you change volume, for example, it will only work for new sound effects that aren't currently playing. I fixed it so it changes the volume of the sounds that are playing currently. That's also the case for disabling sound and also background sound play.

### CODE REVIEW

File: src/Game/Managers/AudioManager.cs [hunk 1]
```diff
@@ -148,6 +163,25 @@ public void UpdateCurrentMusicVolume()
...
+
+            if (Engine.Profile == null || Engine.Profile.Current == null || !Engine.Profile.Current.EnableSound)
+                return;
+
+            float volume = Engine.Profile.Current.SoundVolume / Constants.SOUND_DELTA;
+
+            if (volume < -1 || volume > 1f)
+                return;
+
+            foreach (UOSound sound in _currentSounds.ToArray())
```
Comment (andreakarasho) [inline src/Game/Managers/AudioManager.cs]: Why .ToArray() ?

File: src/Game/Managers/AudioManager.cs [hunk 2]
```diff
@@ -158,6 +192,16 @@ public void StopMusic()
             }
         }
 
+        public void StopSounds()
+        {
+            foreach (UOSound sound in _currentSounds.ToArray())
```
Comment (andreakarasho) [inline src/Game/Managers/AudioManager.cs]: I think you can convert this code to a simple for.

```csharp
for (int i = 0; i < _currentSounds.Count; i++)
{
   var s = _currentSounds[i];
// do stuff
_currentSounds.RemoveAt(i--);
}
```

File: src/Game/Managers/AudioManager.cs [hunk 3]
```diff
@@ -173,6 +217,17 @@ public void Update()
             }
 
             _currentMusic?.Update();
+
+            foreach (UOSound sound in _currentSounds.ToArray())
```
Comment (andreakarasho) [inline src/Game/Managers/AudioManager.cs]: same above

**Review by andreakarasho — COMMENTED** [review]
The most important thing while code is to avoid using LINQ when possible.
The ToArray() method basically create a new Array() filled with the list values in runtime, it's not the best thing to do :)

File: src/Game/Managers/AudioManager.cs (hunk 1 above)
Comment (paezao) [reply src/Game/Managers/AudioManager.cs]: I added that so I could remove from within the loop. But you are right, I will follow your for suggestion.

File: src/Game/Managers/AudioManager.cs (hunk 2 above)
Comment (paezao) [reply src/Game/Managers/AudioManager.cs]: Done

File: src/Game/Managers/AudioManager.cs (hunk 3 above)
Comment (paezao) [reply src/Game/Managers/AudioManager.cs]: Done

File: src/Game/Managers/AudioManager.cs [hunk 4]
```diff
@@ -158,6 +192,17 @@ public void StopMusic()
...
         }
 
+        public void StopSounds()
+        {
+            for (int i = 0; i < _currentSounds.Count; i++)
+            {
+                var sound = _currentSounds[i];
+                sound.Stop();
+                sound.Dispose();
+                _currentSounds.RemoveAt(i);
```
Comment (andreakarasho) [inline src/Game/Managers/AudioManager.cs]: _currentSounds.RemoveAt(i--);

File: src/Game/Managers/AudioManager.cs [hunk 5]
```diff
@@ -173,6 +218,24 @@ public void Update()
...
 
             _currentMusic?.Update();
+
+            for (int i = 0; i < _currentSounds.Count; i++)
+            {
+                var sound = _currentSounds[i];
+
+                if (!sound.IsPlaying())
+                {
+                    _currentSounds.RemoveAt(i);
```
Comment (andreakarasho) [inline src/Game/Managers/AudioManager.cs]: _currentSounds.RemoveAt(i--);

**Review by andreakarasho — COMMENTED** [review]
Don't forget to decrease the index if you remove an element in an array/list while parsing it

**Pull request merged**